assets/js/modules/analytics/components/module/ModuleOverviewWidget/index.js

```diff
@@ -74,29 +75,42 @@ export default function ModuleOverviewWidget( { Widget, WidgetReportError } ) {
 	const statsReport = useSelect( ( select ) => select( MODULES_ANALYTICS ).getReport( statsArgs ) );
 	const statsError = useSelect( ( select ) => select( MODULES_ANALYTICS ).getErrorForSelector( 'getReport', [ statsArgs ] ) );
 
-	if ( overviewError || statsError ) {
-		return <WidgetReportError statsError={ overviewError || statsError } />;
-	}
+	const isZero = isZeroReport( overviewReport );
+	const isError = overviewError || statsError;
 
 	return (
-		<Widget Header={ Header } noPadding>
-			<Overview
-				loaded={ overviewLoaded }
-				report={ overviewReport }
-				selectedStat={ selectedStat }
-				handleStatSelection={ setSelectedState }
-			/>
+		<Widget
+			Header={ Header }
+			noPadding={ ! isError && ! isZero }
+		>
+			{ isError && (
+				<WidgetReportError moduleSlug="analytics" error={ overviewError || statsError } />
+			) }
+			{ ( ! isError && isZero ) && (
+				<WidgetReportZero moduleSlug="analytics" />
+			) }
+			{ ( ! isError && ! isZero ) && (
+				<Fragment>
+					<Overview
+						loaded={ overviewLoaded }
+						report={ overviewReport }
+						selectedStat={ selectedStat }
+						handleStatSelection={ setSelectedState }
+					/>
 
-			<SiteStats
-				loaded={ statsLoaded }
-				selectedStat={ selectedStat }
-				report={ statsReport }
-			/>
+					<SiteStats
+						loaded={ statsLoaded }
+						selectedStat={ selectedStat }
+						report={ statsReport }
+					/>
```

Comment:
I'm not sure how valuable it is to have individual loading states for these two. They just render as a big block anyway, and the only point of having two loaders is that one of them might finish faster - which is not necessarily an advantage: In the All Traffic widget for example we intentionally made it so that all pieces of the widget remain in loading state until everything is done.

Maybe instead of passing a `loaded` prop to `Overview` and `SiteStats`, we could have a single loading state returned by `ModuleOverviewWidget` itself? This would also simplify it and fall nicely in line with my above comment. Basically, there could be another early return for `! overviewLoaded || ! statsLoaded`, which should be wrapped in a `Widget` with the `Header`. This also shouldn't have `noPadding`, which is the case right now for the loading state, making it look different than most our widgets which don't have the `PreviewBlock`s extend to the edges of the widget box.

Reply:
You are right, update!